core/src/main/java/org/bitcoinj/base/AddressParser.java

```diff
@@ -39,6 +39,44 @@ public interface AddressParser {
      */
     Address parseAddress(String addressString, Network network) throws AddressFormatException;
 
+    /**
+     * Parse an address for any known/configured network
+     * @param addressString string representation of address
+     * @return A validated address object
+     * @throws AddressFormatException invalid address string
+     * @deprecated Use {@link #parseAddress(String)}
+     */
+    @Deprecated
+    default Address parseAddressAnyNetwork(String addressString) throws AddressFormatException {
+        return parseAddress(addressString);
+    }
+
+    /**
+     * Functional interface for address parsing. It takes a single parameter, like {@link AddressParser#parseAddress(String)}
+     * but its behavior is context-specific. This interface may be
+     * implemented by creating a partial application of ({@link AddressParser#parseAddress(String, Network)} providing
+     * a fixed value for {@link Network}. Or it may behave more like {@link #parseAddress(String)} with a context-specific
+     * list of networks.
+     */
+    @FunctionalInterface
+    interface Simple {
```

Comment:
I don't get the chosen prefix "Simple". What's more simple about "a context-specific way"?

Reply:
It's "simple" because there's only one method with one parameter. The parent has two methods and a total of three parameters. I'm open to another name, though. Any suggestions?

Reply:
@schildbach Does this make sense? Is there a better name?